editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,356 @@
+                currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+              } else {
+                new Alert(Alert.AlertType.ERROR, "Feature layer failed to load").show();
+              }
+            });
+          } else {
+            new Alert(Alert.AlertType.ERROR, "Unable to get the service feature table").show();
+          }
+        } else {
+          progressIndicator.setVisible(false);
+          new Alert(Alert.AlertType.ERROR, "Service geodatabase failed to load").show();
+        }
+      });
+
+      // listen to clicks on the map to select or move features
+      mapView.setOnMouseClicked(event -> {
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // reset the UI
+          featureLayer.clearSelection();
+          editFeatureVBox.setDisable(true);
+
+          // select the clicked feature
+          Point2D point = new Point2D(event.getX(), event.getY());
+          selectFeature(point);
+        }
+
+        if (event.isStillSincePress() && event.getButton() == MouseButton.SECONDARY) {
+          // if a feature is selected and the current version is not the default, update the feature's geometry
+          if (selectedFeature != null && !serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+            Point2D point = new Point2D(event.getX(), event.getY());
+            Point mapPoint = mapView.screenToLocation(point);
+            selectedFeature.setGeometry(mapPoint);
+            updateFeature(selectedFeature);
+          }
+        }
+      });
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new branch version using user defined values.
+   */
+  @FXML
+  private void handleCreateVersionButtonClicked() {
+
+    // validate version name input
+    if (nameTextField.getText().contains(".") || nameTextField.getText().contains(";") || nameTextField.getText().contains("'") ||
+      nameTextField.getText().contains("\"")) {
+      new Alert(Alert.AlertType.ERROR, "Please enter a valid version name.\nThe name cannot contain the following characters:\n. ; ' \" ").show();
+      return;
+    } else if (nameTextField.getText().length() > 0 && Character.isWhitespace(nameTextField.getText().charAt(0))) {
+      new Alert(Alert.AlertType.ERROR, "Version name cannot begin with a space").show();
+      return;
+    } else if (nameTextField.getText().length() > 62) {
+      new Alert(Alert.AlertType.ERROR, "Version name must not exceed 62 characters").show();
+      return;
+    } else if (nameTextField.getText().length() == 0) {
+      new Alert(Alert.AlertType.ERROR, "Please enter a version name").show();
+      return;
+    }
+
+    // validate version access input
+    if (accessComboBox.getSelectionModel().getSelectedItem() == null) {
+      new Alert(Alert.AlertType.ERROR, "Please select an access level").show();
+      return;
+    }
```

Comment:
If we wanted, we could pre-select one of the three options in this combo box, effectively eliminating the scenario where no Item is selected. Eg. after adding the enum values to the box:
```
accessComboBox.getSelectionModel().select(VersionAccess.PUBLIC);   
```
That way you could get rid of this block entirely.

However I do like the way it is right now, with the prompt 'select', as it doesn't imply any of these options are the 'standard'. So feel free to keep it as is! Just thought it'd be woth mentioning pre-selecting a value.

Reply:
I like the select too, so will leave and see if it comes up again in the next review and get a consensus if needed.